opsdroid/connector/slack/connector.py

```diff
@@ -205,14 +205,23 @@ async def _get_channels(self):
             cursor = None
 
             while True:
-                channels = await self.slack_web_client.conversations_list(
-                    cursor=cursor, limit=self.channel_limit
-                )
-                self.known_channels.update({c["name"]: c for c in channels["channels"]})
-                cursor = channels["response_metadata"].get("next_cursor")
-
-                if not cursor:
-                    break
+                try:
+                    channels = await self.slack_web_client.conversations_list(
+                        cursor=cursor, limit=self.channel_limit
+                    )
+                    self.known_channels.update({c["name"]: c for c in channels["channels"]})
+                    cursor = channels["response_metadata"].get("next_cursor")
+
+                    if not cursor:
+                        break
+                except SlackApiError as error:
```

Comment:
Just a note, we will need to test this change. I was looking at the existing tests and it seems that we didn't do a great job at testing this particular part before your changes 😞 

I haven't tested this, but I **think** you may be able to test the exception by doing something like:

```python

@pytest.mark.asyncio
@pytest.mark.add_response(SlackApiError(...))
async def test_search_history_rate_limit(
    connector, mock_api, caplog
):
    caplog.set_level(logging.INFO)
    await connector.search_history_messages(
        "C01N639ECTY", "1512085930.000000", "1512085980.000000", 1001
    )
    assert mock_api.called("/conversations.history")
    assert "Rate limit threshold reached." in caplog.text
```

You may need to pass things to the SlackApiError, I haven't looked into what they expect. But we can pair on this if you get stuck 😄

Reply:
The API response is this one:
```python
 File "/home/opsdroid_test/.local/lib/python3.10/site-packages/opsdroid/connector/slack/connector.py", line 208, in _get_channels
    channels = await self.slack_web_client.conversations_list(
  File "/home/opsdroid_test/.local/lib/python3.10/site-packages/slack_sdk/web/async_client.py", line 2463, in conversations_list
    return await self.api_call("conversations.list", http_verb="GET", params=kwargs)
  File "/home/opsdroid_test/.local/lib/python3.10/site-packages/slack_sdk/web/async_base_client.py", line 161, in api_call
    return await self._send(
  File "/home/opsdroid_test/.local/lib/python3.10/site-packages/slack_sdk/web/async_base_client.py", line 201, in _send
    return AsyncSlackResponse(**{**data, **res}).validate()
  File "/home/opsdroid_test/.local/lib/python3.10/site-packages/slack_sdk/web/async_slack_response.py", line 192, in validate
    raise e.SlackApiError(message=msg, response=self)
slack_sdk.errors.SlackApiError: The request to the Slack API failed. (url: https://www.slack.com/api/conversations.list)
The server responded with: {'ok': False, 'error': 'ratelimited'}
```

Reply:
I was also able to print out "str(error)" by adding a print statement:
```python
str(error): 
The request to the Slack API failed. (url: https://www.slack.com/api/conversations.list)
The server responded with: {'ok': False, 'error': 'ratelimited'}
```

Reply:
That's great then we can probably just create the `SlackApiError` with that (I'm assuming that the error is a wrapper around the request)